x-pack/platform/plugins/shared/ml/public/application/model_management/models_list.tsx

```diff
@@ -97,28 +97,57 @@ export const getDefaultModelsListState = (): ListingPageUrlState => ({
 });
 
 interface Props {
+  history: ScopedHistory<unknown>;
   pageState?: ListingPageUrlState;
   updatePageState?: (update: Partial<ListingPageUrlState>) => void;
 }
 
-const DOWNLOAD_POLL_INTERVAL = 3000;
-
 export const ModelsList: FC<Props> = ({
   pageState: pageStateExternal,
   updatePageState: updatePageStateExternal,
+  history,
 }) => {
-  const isMounted = useMountedState();
-
   const {
     services: {
-      application: { capabilities },
+      application: { capabilities, navigateToUrl },
       docLinks,
+      overlays,
+      http,
     },
   } = useMlKibana();
 
+  const trainedModelsService = useTrainedModelsService();
+  const isInitialized = trainedModelsService.isInitialized();
+  const items = useObservable(trainedModelsService.models$, trainedModelsService.models);
+  const isLoading = useObservable(trainedModelsService.isLoading$, trainedModelsService.isLoading);
+  const activeOperations = useObservable(
+    trainedModelsService.activeOperations$,
+    trainedModelsService.activeOperations
+  );
+
+  // Navigation blocker when there are active operations
+  useUnsavedChangesPrompt({
```

Comment:
We should allow the user to navigate to another Kibana page, but prompt on closing the Kibana browser tab.